### PR TITLE
Alerts page drop-down filter controls

### LINKED ISSUE #3060: Alerts page drop-down filter controls
## Description 

Drop-down filter controls on Alerts page.

Related:
- https://github.com/elastic/kibana/pull/146989 - initial PR, others also related
- https://github.com/elastic/security-team/issues/4128

| Working Page Filters | When Index is unavailable |
|--|--|
| <video src="https://user-images.githubusercontent.com/7485038/208958595-16eddb1b-cef2-4598-afc2-a69d36ee1d1a.mov" />|Screenshot 2022-12-06 at 13 48 57|

## Notes

- Pushed back to 8.8 release, so creating a separate issue for that.
- Links also added to D&R and Entity Analytics dashboards to jump to Alerts page with these controls turned on and filtering. So we might need to update the docs for those dashboards, since the behavior has changed (and maybe also a new option to jump to Timeline?)
- This feature uses Kibana components, which are documented [here](https://www.elastic.co/guide/en/kibana/current/add-controls.html#create-and-add-options-list-and-range-slider-controls). Those docs assume the controls are in a dashboard, so we can't just point users there; probably just use the Kibana docs as a starting point for creating our own docs specific to Alerts page.

### PR DESCRIPTION
**Pull request by joepeeples**
Resolves #3060.

Previews 

- Text & screenshot updates on **Manage detection alerts** page:
   - [View and filter detection alerts](https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/alerts-ui-manage.html#detection-view-and-filter-alerts) - new bullet (4th from top)
   - [Edit drop-down filter controls](https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/alerts-ui-manage.html#drop-down-filter-controls) - new section
- Screenshot updates:
   - [Visualize detection alerts](https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/visualize-alerts.html)
   - [Visual events analyzer | Find events to analyze](https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/visual-event-analyzer.html#find-events-analyze) - removed image to reduce screenshot maintenance overhead: it's just showing how to enter a query into the KQL bar, doesn't really need to show the whole Alerts page too

### CODE REVIEW

**Comment by logeekal** [comment]
@joepeeples 

This looks good. Thank you.

However, should we add at least one screenshot pointed out what are `drop-down filter controls` here: https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/alerts-ui-manage.html#drop-down-filter-controls

File: docs/detections/alerts-ui-manage.asciidoc [hunk 1]
```diff
@@ -37,6 +39,41 @@ image::images/additional-filters.png[Alerts table with Additional filters menu h
...
+* You can have a maximum of four controls on the Alerts page.
+* You can't remove the *Status* control.
+* If you make any changes to the controls, you _must_ save the pending changes in order for them to persist.
+====
+
+. Click the three-dot icon next to the controls (image:images/three-dot-icon-blue.png[Three-dot icon,18,18]), then select *Edit Controls*.
+
+. Do any of the following:
+
+* To reorder controls, click and drag a control by the handle (image:images/handle-icon.png[Control handle,18,18]).
```
Comment (benironside) [inline docs/detections/alerts-ui-manage.asciidoc:59]:
```suggestion
* To reorder controls, click and drag a control by its handle (image:images/handle-icon.png[Control handle,18,18]).
```

**Review by benironside — COMMENTED** [review]
LGTM, but I'm left wondering whether the order of the filters matters, since you describe how to reorder them. I don't see how it would matter, but then what's the point of reordering them. Perhaps you could add some clarity around this

File: docs/detections/alerts-ui-manage.asciidoc [hunk 2]
```diff
@@ -21,6 +21,8 @@ image::images/view-alert-details.png[View details button, 200]
 
 * Use the date and time filter to define a specific time range. By default, this filter is set to search the last 24 hours.
 
+* Use the drop-down filter controls to filter alerts by up to four fields. By default, you can filter by *Status*, *Severity*, *User*, and *Host*, and you can <<drop-down-filter-controls,edit the controls>> to use other fields.
```
Comment (nastasha-solomon) [inline docs/detections/alerts-ui-manage.asciidoc:24]:
```suggestion
* Use the drop-down filter controls to filter alerts by up to four fields. By default, you can filter alerts by *Status*, *Severity*, *User*, and *Host*, and you can <<drop-down-filter-controls,edit the controls>> to use other fields.
```

File: docs/detections/alerts-ui-manage.asciidoc [hunk 3]
```diff
@@ -37,6 +39,41 @@ image::images/additional-filters.png[Alerts table with Additional filters menu h
...
+
+* To reorder controls, click and drag a control by the handle (image:images/handle-icon.png[Control handle,18,18]).
+
+* To remove a control, hover over it and select *Remove control* (image:images/red-x-icon.png[Red X icon,18,18]).
+
+* To edit a control, hover over it and select *Edit control* (image:images/pencil-icon.png[Edit icon,18,18]).
+
+* To add a new control, click *Add Controls* (image:images/add-icon-blue.png[Add icon,18,18]). If you already have four controls, you must first remove one to make room for the new one.
+
+. If you're editing or adding a control, in the configuration flyout that opens:
```
Comment (nastasha-solomon) [inline docs/detections/alerts-ui-manage.asciidoc:67]:
```suggestion
. If you're editing or adding a control, do the following in the configuration flyout that opens:
```

**Review by nastasha-solomon — APPROVED** [review]
Looks great!

**Comment by joepeeples** [comment]
> LGTM, but I'm left wondering whether the order of the filters matters, since you describe how to reorder them. I don't see how it would matter, but then what's the point of reordering them. Perhaps you could add some clarity around this

I think the order is just a user preference; I can add something to indicate that.

File: docs/detections/alerts-ui-manage.asciidoc [hunk 4]
```diff
@@ -37,6 +39,41 @@ image::images/additional-filters.png[Alerts table with Additional filters menu h
...
+[[drop-down-filter-controls]]
+=== Edit drop-down filter controls
+
+By default, the drop-down controls on the Alerts page filter alerts by *Status*, *Severity*, *User*, and *Host*. You can edit them to filter by different fields, as well as remove, add, and reorder them.
+
+[NOTE]
+====
+* You can have a maximum of four controls on the Alerts page.
+* You can't remove the *Status* control.
+* If you make any changes to the controls, you _must_ save the pending changes in order for them to persist.
```
Comment (jmikell821) [inline docs/detections/alerts-ui-manage.asciidoc:52]:
```suggestion
* If you make any changes to the controls, you _must_ save the pending changes for them to persist.
```

File: docs/detections/alerts-ui-manage.asciidoc [hunk 5]
```diff
@@ -102,7 +139,7 @@ From the Alerts table or the alert details flyout, you can:
 
 You can set an alert's status to indicate whether it needs to be investigated
 (*Open*), is under active investigation (*Acknowledged*), or has been resolved
-(*Closed*). By default, the Alerts page displays open alerts. To filter alerts that are *Acknowledged* or *Closed*, select the appropriate status in the upper-left corner of the Alerts page. This filters all of the results displayed on the page (the Count table, Trend histogram, and Alerts table).
+(*Closed*). By default, the Alerts page displays open alerts. To filter alerts that are *Acknowledged* or *Closed*, use the *Status* drop-down filter at the top of the Alerts page. This filters all of the results displayed on the page.
```
Comment (jmikell821) [inline docs/detections/alerts-ui-manage.asciidoc:142]: I don't think the last sentence is needed as it should be obvious. 

```suggestion
(*Closed*). By default, the Alerts page displays open alerts. To filter alerts that are *Acknowledged* or *Closed*, use the *Status* drop-down filter at the top of the Alerts page. 
```

**Review by jmikell821 — COMMENTED** [review]
Just some tiny nits. Looks awesome! 🎉

**Comment by joepeeples** [comment]
> However, should we add at least one screenshot pointing out what are `drop-down filter controls` here: https://security-docs_3256.docs-preview.app.elstc.co/guide/en/security/master/alerts-ui-manage.html#drop-down-filter-controls

@logeekal Added screenshot in https://github.com/elastic/security-docs/pull/3256/commits/3c9fc630e6abb39ef8d191c350e3fbf4db838c0f, thanks for the suggestion!

**Pull request merged**